scripts/src/quay/linkimages.py

```diff
@@ -0,0 +1,133 @@
+"""
+Used by a github action to link a newly published chart verifier image to a specified tag in quay
+Can also be called from the command line.
+
+parameters:
+    ---link-tag : tag to link the image to, default is "test"
+    --verifier-version : the version of the new chart verifier release
+
+notes:
+    It is anticipated that the github action will invoke this function just before the new image is available
+    in quay. As a result the function will look for the image every 15 seconds for up to 15 minutes.
+
+    To invoke from the command line, invoke from the root directory of the repository,for example:
+          python3 scripts/src/quay/linkimages.py --verifier-version=1.2.0.
+
+    If the verifier version image is already linked to the specified tag no action is taken.
+
+    The default of link-tag is set to "test" to avoid accidental updates to latest.
+
+    The auth token to enable the script to link tags in quay must be set in an environment variable QUAY_AUTH_TOKEN.
+    For a github action this must be set as a repository secret.
+    For information on auth token required see:
+    https://access.redhat.com/documentation/en-us/red_hat_quay/3/html/red_hat_quay_api_guide/using_the_red_hat_quay_api
+
+results:
+    A message indicating the outcome.
+    exit code 1 if version image was not found.
+
+"""
+
+import requests
+import json
+import sys
+import argparse
+import os
+from retry import retry
+
+sys.path.append('./scripts/src/')
+from release import releasechecker
+
+# Quay API docs: https://access.redhat.com/documentation/en-us/red_hat_quay/3/html/red_hat_quay_api_guide/index
+# Quay Swagger API: https://docs.quay.io/api/swagger/
+#
+
+defaultLinkTag= "test"
+tagUrl = 'https://quay.io/api/v1/repository/redhat-certification/chart-verifier/tag/'
+
+# try every 15 seconds for 15 minutes
+@retry(Exception,tries=60, delay=15)
+def getImageId(tagValue,doRetry):
+
+    print(f"[INFO] look for tag : {tagValue}, retry : {doRetry}")
+    tagUrl = 'https://quay.io/api/v1/repository/redhat-certification/chart-verifier/tag/'
+
+    get_params = {'onlyActiveTags' : 'true','specificTag' : tagValue }
+
+    response = requests.get(tagUrl,params=get_params)
+
+    imageId = ""
+    if response.status_code > 201:
+        print(f"[Error] Error getting tags from quay : status_code={response.status_code}")
+    else:
+        tags = json.loads(response.text)
+        print("[INFO] loaded the tags")
+        for tag in tags["tags"]:
+            if tag['name'] == tagValue:
+                imageId = tag['image_id']
+                print(f"[INFO] Found tag {tagValue}. image_id : {imageId}")
+                break
+            else:
+                print(f"[INFO] ignore tag {tag['name']}")
+
+        if not imageId and doRetry:
+            print(f"[INFO] {tagValue} not found. Retry!")
+            raise Exception(f"Image {tagValue} not found")
+
+    return imageId
```

Comment:
Do you want to return empty on error? If you do then why printing error and not warning?

Reply:
The method is used to look for the latest tag as well as the new n.n.n tag. When looking up latest it would not be be error if it did not exist yet. I did add some logic to the caller to error if an empty string is returned for the n.n.n tag although in theory that should always result in an exception.